atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -100,12 +102,59 @@
 import com.palantir.lock.impl.LegacyTimelockService;
 import com.palantir.lock.impl.LockServiceImpl;
 import com.palantir.lock.v2.TimelockService;
-import com.palantir.logsafe.UnsafeArg;
 import com.palantir.timestamp.TimestampService;
 import com.palantir.timestamp.TimestampStoreInvalidator;
 import com.palantir.util.OptionalResolver;
 
-public final class TransactionManagers {
+@Value.Immutable
+public abstract class TransactionManagers {
+    abstract AtlasDbConfig config();
+
+    @Value.Default
+    Supplier<Optional<AtlasDbRuntimeConfig>> runtimeConfigSupplier() {
+        return Optional::empty;
+    }
+
+    abstract Set<Schema> schemas();
+
+    @Value.Default
+    Consumer<Object> env() {
+        return resource -> { };
+    }
+
+    @Value.Default
+    LockServerOptions lockServerOptions() {
+        return LockServerOptions.DEFAULT;
+    }
+
+    @Value.Default
+    boolean allowHiddenTableAccess() {
+        return false;
+    }
+
+    abstract Optional<Class<?>> callingClass();
+
+    abstract Optional<String> userAgent();
+
+    // directly specified -> inferred from caller -> default
+    @Value.Derived
+    String derivedUserAgent() {
+        return userAgent().orElse(callingClass().map(UserAgents::fromClass).orElse(UserAgents.DEFAULT_USER_AGENT));
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder extends ImmutableTransactionManagers.Builder {
+        public SerializableTransactionManager buildSerializable() {
+            return build().serializable();
+        }
```

Comment:
do we need to expose `buildSnapshot`?

Reply:
we could? this got me to feature parity

Reply:
yeah I think we're good with just `Serializable`. that's all the current API gives you